addons/payment/models/account_journal.py

```diff
@@ -12,3 +12,12 @@ def _check_journal_type_change(self):
         acquirer_incompatible_journals = self.filtered(lambda j: j.type not in ('bank', 'cash'))
         if acquirer_incompatible_journals and self.env['payment.acquirer'].search_count([('journal_id', 'in', acquirer_incompatible_journals.ids)]):
             raise ValidationError(_("An acquirer is using this journal. Only bank and cash types are allowed."))
+
+    # prevent the deletion of account journals used in any payment acquirer
+    def unlink(self):
+        linked_payment_acquirers = self.env['payment.acquirer'].sudo().search([]).filtered_domain(
+                [('journal_id', 'in', self.ids)])
```

Comment:
```suggestion
        if not self:
            return
        linked_payment_acquirers = self.env['payment.acquirer'].sudo().search([
            ('state', 'in', ['test', 'enabled']),
            ('journal_id', 'in', self.ids),
        ])
```

Reply:
here search wouldn't work on journal_id because it is a computed non-stored field that is why I did it this way

Reply:
https://github.com/odoo/odoo/blob/b40e0ea6771f31fa1ecd392b543a47f45950cd00/addons/payment/models/payment_acquirer.py#L111-L113

Please look at 14.0 code when you fix things in 14.0 ...